server/channels/app/scheduled_post.go

```diff
@@ -11,6 +11,11 @@ import (
 )
 
 func (a *App) SaveScheduledPost(rctx request.CTX, scheduledPost *model.ScheduledPost) (*model.ScheduledPost, *model.AppError) {
+	scheduledPost.PreSave()
+	if validationErr := scheduledPost.BaseIsValid(); validationErr != nil {
```

Comment:
We check it in store level, not app level. If you see the IsValid function, it accepts the allowed max length of message, which is available only in the SQL store layer because its computed by querying the database for max allowed length of message column in the database. So, on app layer I'm validating everything except the message length, and message length is being checked on store layer along with all other checks (yes its duplicate check but its fine).

Previous all checks were only at store layer, causing a failure to communicate the validation error in API responses, so I moved all validations minus message length here. This is an improvement how we validate drafts and posts.

Reply:
Ah, I see. In that case, perhaps instead of splitting up the method, we could just pass some arbitrarily large number here and comment that it's double checked later? 0/5

Reply:
I don't think it provides any benefit and the check will obviously wrong, so even if isValid returned true in app layer, you still aren't sure if the entire object is valid, kinda faking the meaning of Isvalid. I'd prefer to not do it.